event/datadeal/data_verification_event.go

```diff
@@ -1 +1,216 @@
 package datadeal
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/config"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/ipfs"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/medibloc/panacea-doracle/validation"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+var _ event.Event = (*DataVerificationEvent)(nil)
+
+type DataVerificationEvent struct {
+	reactor reactor
+}
+
+type reactor interface {
+	GRPCClient() *panacea.GrpcClient
+	EnclaveInfo() *sgx.EnclaveInfo
+	OracleAcc() *panacea.OracleAccount
+	OraclePrivKey() *btcec.PrivateKey
+	Config() *config.Config
+	QueryClient() *panacea.QueryClient
+	Ipfs() *ipfs.Ipfs
+}
+
+func NewDataVerificationEvent(r reactor) DataVerificationEvent {
+	return DataVerificationEvent{r}
+}
+
+func (d DataVerificationEvent) GetEventType() string {
+	return "message"
+}
+
+func (d DataVerificationEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (d DataVerificationEvent) GetEventAttributeValue() string {
+	return "'SellData'"
+}
+
+func (d DataVerificationEvent) EventHandler(event ctypes.ResultEvent) error {
+	dealIDStr := event.Events[types.EventTypeDataVerificationVote+"."+types.AttributeKeyDealID][0]
+	dataHash := event.Events[types.EventTypeDataVerificationVote+"."+types.AttributeKeyDataHash][0]
+
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	deal, err := d.reactor.QueryClient().GetDeal(dealID)
+	if err != nil {
+		return err
+	}
```

Comment:
If error is `NotFound`, vote `NO`.
Otherwise, return error and do nothing.

I think it's impossible to sell the data to non-existing deal, but it make sense to vote `NO` with non-existing deal.

Reply:
Thanks for comment. I have a question. How about change a return type of `EventHandler` method of `Event` interface to `types.VOTE_OPTION`?